src/components/linter.ts

```diff
@@ -1,248 +1,68 @@
 import * as vscode from 'vscode'
-import * as path from 'path'
-import * as fs from 'fs'
-import * as os from 'os'
-import {ChildProcessWithoutNullStreams, spawn} from 'child_process'
-import {EOL} from 'os'
 
 import type {Extension} from '../main'
+import { ChkTeX } from './linterlib/chktex'
+import { LaCheck } from './linterlib/lacheck'
+export interface ILinter {
+    readonly linterDiagnostics: vscode.DiagnosticCollection,
+    lintRootFile(): void,
+    lintFile(document: vscode.TextDocument): void,
+    parseLog(log: string, filePath?: string): void
+}
 
 export class Linter {
-    private readonly extension: Extension
+    readonly #linters: Map<string, ILinter> = new Map()
     private linterTimeout?: NodeJS.Timer
-    private readonly currentProcesses = Object.create(null) as { [linterId: string]: ChildProcessWithoutNullStreams }
 
-    constructor(extension: Extension) {
-        this.extension = extension
-    }
+    constructor(private readonly extension: Extension) {}
 
-    private get rcPath() {
-        let rcPath: string
-        // 0. root file folder
-        const root = this.extension.manager.rootFile
-        if (root) {
-            rcPath = path.resolve(path.dirname(root), './.chktexrc')
-        } else {
-            return
-        }
-        if (fs.existsSync(rcPath)) {
-            return rcPath
+    private get chktex(): ChkTeX {
+        const linterId = 'chktex'
+        if (!this.#linters.has(linterId)) {
+            this.#linters.set(linterId, new ChkTeX(this.extension))
         }
+        return this.#linters.get(linterId) as ChkTeX
```

Comment:
Please remove `as ChkTeX`. Please stop using type assertions casually. When you use type assertions, it is just wrong generally.

If you don't know how to fix it, please use `#chktex` as I suggested.